Automattic/jetpack
# AI Client: Add Markdown and HTML conversions

Fixes #36905

## Proposed changes:

* Extracts Markdown conversions to a separated lib on the AI Client js-package
* Fixes some small typescript issues

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






Problems should be easy to spot
* Make a request on the AI Assistant
* Request an excerpt
* Request a feedback
* Use the voice to content block

Content should be converted to and from markdown as expected
